policyengine_us/variables/gov/states/nj/tax/income/nj_income_tax_before_refundable_credits.py

```diff
@@ -8,3 +8,28 @@ class nj_income_tax_before_refundable_credits(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.NJ
+
+    def formula(tax_unit, period, parameters):
+        taxable_income = tax_unit("nj_taxable_income", period)
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+
+        # Get main nj tax parameter tree.
+        p = parameters(period).gov.states.nj.tax.income.main
+
+        return select(
```

Comment:
FYI this could be this if not for my suggestion to make one per filing status:
```python
return where((filing_status == status.SINGLE) | (filing_status == status.SEPARATE), p.single_separate.calc(taxable_income), p.joint_hoh_spouse.calc(taxable_income))
```